### ISSUE TITLE
suggestions for improvements

### ISSUE THREAD

**Issue by wordiboi** [Amberg/DocxTemplater#5]
Hi,

first of all thank you very much for this amazing project. I've been trying to do something similiar a few years back but then decided to use some existing "mail merge" library. I've been trying your library and I really love it, however I got some ideas for improvement:

**1. Give users an option to decide whether applying the template will throw an exception or not when a certain property does not exist in your model.**

Right now it throws an error whenever a property does not exist. I think there should be an option where either the no existing property either gets replace by a whitespace or you leave the template syntax variable as it is without replacing it.

**2. Give users an option to choose whether e.g. loop start and end tags create an "empty paragraph" in the final .docx file or if those empty paragraphs get removed (this might also be thing regarding nr. 1 when a non existing variable get replaced by "nothing").
Right now this template (without using a table)**
```
Text123
{{#ds.Items}}
{{Items.Name}} {{Items.Price < 6}} less than 6 {{else}} more than 6{{/}}
{{/ds.Items}}
Text456
```
will create the following result:
```
Text123
.
ItemName less than 6
.
ItemName more than 6
.
Text456
```
It might also be pretty neat and cause less clutter if inside a loop you were able to just reference the corresponding property without reference the object itself (e.g. "Items") once again, like:

```
{{Items}} 
{{Name}} {{Price}}
{{/Items}}
```
but it guess this might get tricky when another property with the same name exists outside the loop object.

**3. Is it nescessary to bind a prefix to the model?**

Right now without using the prefix conditionals will only work inside loops while the following conditional wont work outside:
`{{Test > 5}}success{{/}}`
but it works eventually when using the prefix like:
`{{ds.Test > 5}}success{{/}}`

This is somehow confusing regarding the fact, that "regular" variables like {{Title}} will work without using the model's prefix.

**4. It would probably also be helpful (atleast for readability) to allow leading and trailing whitespaces inside the bracket syntax and instead use something like Trim() to remove those whitspaces internally.**

Right now {{ Title }} won't work while {{Title}} will.

I'm also trying to give some input, but I've never worked DynamicExpresso before, therefore I've to get into all of this first.

Thanks again for the great work.

**Comment by Amberg** [comment]
Thank you for your feedback. I really appreciate that, that's exactly what open source is all about.

> 1. Give users an option to decide whether applying the template will throw an exception or not when a certain property does not exist in your model.

This was already on my wish / todo list and I've now implemented this. In the current implementation, the placeholder is removed in the event of an error if configured in this way. I wonder if there should perhaps also be the possibility to output the errors in the document

> . Give users an option to choose whether e.g. loop start and end tags create an "empty paragraph"
The paragraph of the placeholder should be removed in any case. If the user want's a paragraph he can insert a paragraph after or before the placeholder. 

The other points have also been fixed

> It might also be pretty neat and cause less clutter if inside a loop you were able to just reference the corresponding property without reference the object itself

This is not possible because there can be nested loops or a variable with the same name. 
I have already thought about introducing a syntax that can define a loop variable..
`{{#Items#i}}`...{{i.Value}} .. {{/}}
